docs/about-core-concepts/architecture-guide.rst

```diff
@@ -22,20 +22,24 @@ be indexed into the ODC, so long as it can be described by ODC metadata.
 This section of the documentation describes the structure of the ODC and the key
 components that make up an implementation, as well as the ecosystem around it.
 
+.. figure:: ../diagrams/odc1.9-arch.png
```

Comment:
I think this figure is a good start, but wonder if it's too complex to match the content/goal of this page, which should probably be a more general overview. There are multiple elements in the diagram that have not been explained in the text, so I think this figure might confuse people. I wonder if we need a more technical architecture page, which serves as a reference, and that this page should focus more on explanation -- see https://www.diataxis.fr/reference-explanation/

Reply:
Yeah, it's a bit of mish-mash of multiple persepectives, but at least (in contrast to the diagram it replaces) it actually captures some relevant architectural information!